department-of-veterans-affairs/abd-vro
# Add VRO Console container

## What was the problem?


With VRO's deployment to prod, we'll need ways to investigate and recover from errors in the production environment.

When an error occurs, to investigate and diagnose, we'll need to examine the processing state of the claim by looking at Camel routes and the DB contents.

Associated ticket: #502 

## How does this fix it?[^1]


Create a "VRO Console" container that connects to the DB to allow investigation of DB contents.
To keep this relatively PR small, the following will be in separate PRs:
- Connecting with Camel within the Console.
- Adding this console container to deploy scripts and GH Actions
- Tests for the new code
- Add Groovy test code coverage check
- ~Remove `LABEL` command from all `Dockerfile`s~ (SecRel uses the Dockerfile directly, so let's keep the LABEL)

## How to test this PR
- `export COMPOSE_PROFILES=debug` -- the console container only starts if the docker-compose `debug` profile is enabled.
- start VRO. Note the new `vro-console-1` container; check out the logs: `docker logs vro-console-1`.
- Attach to the container: `docker attach vro-console-1`
- Review https://groovy-lang.org/groovysh.html
- On the `groovy:000>` prompt, try the following:
```groovy
?  # display help
:show variables
claimsT.findAll().collect{ it.claimSubmissionId }
c = getAnyClaim()
// Different ways to do the same thing:
printJson c
printJson getAnyClaim()
pj c
// exit   # This will stop the container
// Instead, press Ctrl-p Ctrl-q to detach from the container without stopping it
```

If you need to restart the console container:
```
cd app/src/docker
docker-compose up -d console
docker attach vro-console-1
```

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
